lib/trento/infrastructure/software_updates/adapter/suma_http_executor.ex

```diff
@@ -208,21 +218,10 @@ defmodule Trento.Infrastructure.SoftwareUpdates.Suma.HttpExecutor do
   defp ssl_options(nil), do: []
 
   defp ssl_options(ca_cert),
-    do: [ssl: [verify: :verify_peer, cacerts: [get_cert_der(ca_cert)]]]
-
-  def get_cert_der(ca_cert) do
-    {cert_type, cert_entry} =
-      ca_cert
-      |> :public_key.pem_decode()
-      |> hd()
-      |> :public_key.pem_entry_decode()
-      |> split_type_and_entry()
-
-    :public_key.der_encode(cert_type, cert_entry)
-  end
+    do: [ssl: [verify: :verify_peer, cacerts: get_cert_der(ca_cert)]]
 
-  def split_type_and_entry(ans1_entry) do
-    ans1_type = elem(ans1_entry, 0)
-    {ans1_type, ans1_entry}
+  defp split_type_and_entry(asn1_entry) do
+    asn1_type = elem(asn1_entry, 0)
+    {asn1_type, asn1_entry}
```

Comment:
nitpick: this function could be just declarative:

```suggestion
  defp split_type_and_entry({asn1_type, _} = asn1_entry) do
    {asn1_type, asn1_entry}
```

Reply:
At the end I removed the function altogether and moved it in the mapping pipe 9569dde82b7363b1dbce568f06e962620bbea11b 

Just for reference, the pem decoded entry seems to be a 4 elemet tuple rather than a 2 element tuple https://www.erlang.org/doc/apps/public_key/public_key.html#t:pem_entry/0.

Hope it remains like this 😄